apps/web/src/pages/templates/components/formTypes.ts

```diff
@@ -28,19 +33,58 @@ export interface IStepEntity {
     value?: BuilderGroupValues;
     children?: FilterParts[];
   }[];
-  active: boolean;
-  shouldStopOnFail: boolean;
+  active?: boolean;
+  shouldStopOnFail?: boolean;
   replyCallback?: {
     active: boolean;
     url?: string;
   };
-  metadata?: {
-    amount?: number;
-    unit?: DigestUnitEnum;
-    type?: DigestTypeEnum;
+  digestMetadata?: {
+    type: DigestTypeEnum;
     digestKey?: string;
-    delayPath?: string;
-    backoff?: boolean;
+    [DigestTypeEnum.REGULAR]?: {
+      amount: string;
+      unit: DigestUnitEnum;
+      backoff?: boolean;
+      backoffAmount?: number;
+      backoffUnit?: DigestUnitEnum;
+    };
+    [DigestTypeEnum.TIMED]?: {
+      unit: DigestUnitEnum;
+      [DigestUnitEnum.MINUTES]?: {
+        amount: string;
+      };
+      [DigestUnitEnum.HOURS]?: {
+        amount: string;
+      };
+      [DigestUnitEnum.DAYS]?: {
+        amount: string;
+        atTime: string;
+      };
+      [DigestUnitEnum.WEEKS]?: {
+        amount: string;
+        atTime: string;
+        weekDays: DaysEnum[];
+      };
+      [DigestUnitEnum.MONTHS]?: {
+        amount: string;
+        atTime: string;
+        monthDays: number[];
+        monthlyType: MonthlyTypeEnum;
+        ordinal?: OrdinalEnum;
+        ordinalValue?: OrdinalValueEnum;
+      };
+    };
+  };
```

Comment:
This is the digest metadata, the idea here is to have a separate state for the regular/timed digest values. Also the timed digest also has an independent state for the minutes, hours, days etc. This allows us to keep the form input values "persisted" between switching the tabs: regular/timed, minutes/hours/days/etc.